pkg/util/predicates.go

```diff
@@ -71,6 +73,50 @@ func (p FinalizersChangedPredicate) Update(e event.UpdateEvent) bool {
 		!reflect.DeepEqual(e.MetaOld.GetFinalizers(), e.MetaNew.GetFinalizers())
 }
 
+// FilterForOwner will only allow events that owned by noobaa
+type FilterForOwner struct {
+	OwnerSubstrings []string
+}
+
+// hasCorrectOwner checks if one of the owners has a substring that represents an expected owner
+func hasCorrectOwner(arr []v1.OwnerReference, ownerSubstrings []string) bool {
+	for _, b := range arr {
+		for _, sub := range ownerSubstrings {
+			if strings.Contains(b.Name, sub) {
+				return true
+			}
+		}
+	}
+	return false
+}
```

Comment:
Wouldn't it be easier to have it as a `FilterForOwner` method? you will have access to the `OwnerSubstrings` via object `r.OwnerSubstrings` and will not pollute the package scope

Reply:
thanks, fixed.